lib/sqlalchemy/orm/mapped_collection.py

```diff
@@ -153,9 +184,11 @@ def column_keyed_dict(
         coercions.expect(roles.ColumnArgumentRole, q, argname="mapping_spec")
         for q in util.to_list(mapping_spec)
     ]
-    keyfunc = _PlainColumnGetter(cols)
-    return _mapped_collection_cls(
-        keyfunc, ignore_unpopulated_attribute=ignore_unpopulated_attribute
+    keyfunc = _PlainColumnGetter(cols)  # type: ignore
```

Comment:
i want to see if we don't need four lines of ignore here, this suggests we aren't hitting the mark...

Reply:
I've removed all the ignores from here, could you please check?